presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/WindowNodeUtil.java

```diff
@@ -75,4 +81,20 @@ public static BoundType toBoundType(FrameBound.Type type)
                 throw new UnsupportedOperationException(format("unrecognized frame bound type %s", type));
         }
     }
+
+    // Explicitly limit the following functions for WindowNode.
+    // TODO: Once the arguments in CallExpression are pure RowExpressions, move the method to SymbolsExtractor
+    public static Set<Symbol> extractWindowFunctionUnique(WindowNode.Function function)
+    {
+        ImmutableSet.Builder<Symbol> builder = ImmutableSet.builder();
+        for (RowExpression argument : function.getFunctionCall().getArguments()) {
+            if (isExpression(argument)) {
+                builder.addAll(SymbolsExtractor.extractAll(castToExpression(argument)));
+            }
+            else {
+                builder.addAll(SymbolsExtractor.extractAll(argument).stream().map(variable -> new Symbol(variable.getName())).collect(toSet()));
```

Comment:
I guess this doesn't really matter but we normally use `toImmutableSet` so just keep it consistent?